pandas/tests/io/pytables/test_complex.py

```diff
@@ -0,0 +1,193 @@
+from warnings import catch_warnings
+
+import numpy as np
+import pytest
+
+import pandas.util._test_decorators as td
+
+import pandas as pd
+from pandas import DataFrame, Series
+from pandas.tests.io.pytables.common import ensure_clean_path, ensure_clean_store
+import pandas.util.testing as tm
+from pandas.util.testing import assert_frame_equal
+
+from pandas.io.pytables import read_hdf
+
+tables = pytest.importorskip("tables")
```

Comment:
right can you move these to common.py (the tables.parameters.*) that are repeated across files.

you can also move the pytest.importorskip to common.py as well

Reply:
But this would mean having to import pytest in your common module, is this something you'd be ok with?

Reply:
@jreback would this be acceptable?

```diff
diff --git a/pandas/tests/io/pytables/conftest.py b/pandas/tests/io/pytables/conftest.py
index 6164f5d07..db00b1e33 100644
--- a/pandas/tests/io/pytables/conftest.py
+++ b/pandas/tests/io/pytables/conftest.py
@@ -15,3 +15,14 @@ def setup_mode():
     tm.reset_testing_mode()
     yield
     tm.set_testing_mode()
+
+
+@pytest.fixture(scope="module", autouse=True)
+def set_parameters():
+    """
+    Set pytables parameters so we don't have file sharing.
+    """
+    tables = pytest.importorskip("tables")
+    tables.parameters.MAX_NUMEXPR_THREADS = 1
+    tables.parameters.MAX_BLOSC_THREADS = 1
+    tables.parameters.MAX_THREADS = 1
diff --git a/pandas/tests/io/pytables/test_complex.py b/pandas/tests/io/pytables/test_complex.py
index 96c4a3042..37739b1fd 100644
--- a/pandas/tests/io/pytables/test_complex.py
+++ b/pandas/tests/io/pytables/test_complex.py
@@ -14,10 +14,6 @@ from pandas.util.testing import assert_frame_equal
 from pandas.io.pytables import read_hdf
 
 tables = pytest.importorskip("tables")
-# set these parameters so we don't have file sharing
-tables.parameters.MAX_NUMEXPR_THREADS = 1
-tables.parameters.MAX_BLOSC_THREADS = 1
-tables.parameters.MAX_THREADS = 1
 
 
 # GH10447
diff --git a/pandas/tests/io/pytables/test_store.py b/pandas/tests/io/pytables/test_store.py
index 97fd89903..e31cb0cf6 100644
--- a/pandas/tests/io/pytables/test_store.py
+++ b/pandas/tests/io/pytables/test_store.py
@@ -56,10 +56,6 @@ from pandas.io.pytables import TableIterator  # noqa: E402 isort:skip
 
 
 tables = pytest.importorskip("tables")
-# set these parameters so we don't have file sharing
-tables.parameters.MAX_NUMEXPR_THREADS = 1
-tables.parameters.MAX_BLOSC_THREADS = 1
-tables.parameters.MAX_THREADS = 1
 
 
 _default_compressor = "blosc"
diff --git a/pandas/tests/io/pytables/test_timezones.py b/pandas/tests/io/pytables/test_timezones.py
index 3e91c6381..b18fdc7b6 100644
--- a/pandas/tests/io/pytables/test_timezones.py
+++ b/pandas/tests/io/pytables/test_timezones.py
@@ -16,10 +16,6 @@ import pandas.util.testing as tm
 from pandas.util.testing import assert_frame_equal, set_timezone
 
 tables = pytest.importorskip("tables")
-# set these parameters so we don't have file sharing
-tables.parameters.MAX_NUMEXPR_THREADS = 1
-tables.parameters.MAX_BLOSC_THREADS = 1
-tables.parameters.MAX_THREADS = 1
 
 
 def _compare_with_tz(a, b):

```

Reply:
the other alternative is to monkeypatch..

```python
@pytest.fixture(autouse=True)
def set_parameters(monkeypatch):
    """
    Set pytables parameters so we don't have file sharing.
    """
    monkeypatch.setattr("tables.parameters.MAX_NUMEXPR_THREADS", 1, raising=False)
    monkeypatch.setattr("tables.parameters.MAX_BLOSC_THREADS", 1, raising=False)
    monkeypatch.setattr("tables.parameters.MAX_THREADS", 1, raising=False)
```

but this needs to be a function scoped fixture, see https://github.com/pytest-dev/pytest/issues/363

Reply:
in theory it would be better as a fixture but just trying to avoid code duplication

this was always somewhat leaky because it globally patches

it could be nicer but can do later

Reply:
ok. but IMO common.py should not have pytest imported.

also, some of the functions in common could be fixturized (and maybe eventually remove common.py) 

i'm not convinced from the pytest docs that pytest.importorskip should be used outside the module, but it seems to work.

but we need to merge ASAP to avoid merge conflicts, so i'll not block this further.

Reply:
so what is the verdict on tables for this PR?

Reply:
> so what is the verdict on tables for this PR?

see https://github.com/pandas-dev/pandas/pull/28746#discussion_r331204204